files/epilog.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+#
+# Copyright 2018-2020 Ghent University
+#
+# This file is part of slurm-prepilogue
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# All rights reserved.
+#
+# #
+
+HERE=$(dirname "$0")
+LOGGER=/usr/bin/logger
+
+
+function user_ipcs_cleanup(){
+    # Called at end of last job by user to:
+    #  * shared memory segments that have no processes attached
+
+    ${LOGGER} -p local0.alert "Starting user_ipcs_cleanup() for ${SLURM_JOB_USER} from ${SLURM_JOB_ID}"
+
+    # kill leftover allocated shared memory
+    for S in $(/usr/bin/ipcs -m | awk -v user=${SLURM_JOB_USER} ' \
+            /^key/ { \
+                for (i=1; i<=NF; i++) { \
+                    f[$i] = i \
+                } \
+            } \
+            { \
+                if ($f["owner"] == user && $f["nattch"] == 0) { \
+                    print $f["shmid"]; \
+                } \
+            }' \
+    ); do
+        /usr/bin/ipcrm -s $S
+    done
+
+    ${LOGGER} -p local0.alert "Finished user_ipcs_cleanup() for ${SLURM_JOB_USER} from ${SLURM_JOB_ID}"
+}
+
+if [ x$SLURM_UID = "x" ] ; then
+    exit 0
+fi
+if [ x$SLURM_JOB_ID = "x" ] ; then
+    exit 0
+fi
+
+
+${LOGGER} -p local0.alert "********  starting $0 for job $SLURM_JOB_ID"
+
+#
+# Only run for valid VSC user IDs
+#
+if [ $SLURM_UID -lt 2500000 ] ; then
```

Comment:
what is this supposed to catch? and if the docs are right, should it nog be higher than a certain number as well? like 1000 or something?

Reply:
VSC user IDs start at 2510001. Not sure what you mean.